core/src/avm1/globals/as_broadcaster.rs

```diff
@@ -0,0 +1,249 @@
+    ) {
+        initialize_internal(gc_context, broadcaster, self, array_proto);
+    }
+}
+
+pub fn add_listener<'gc>(
+    activation: &mut Activation<'_, 'gc>,
+    context: &mut UpdateContext<'_, 'gc, '_>,
+    this: Object<'gc>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error<'gc>> {
+    let new_listener = args.get(0).cloned().unwrap_or(Value::Undefined);
+    let listeners = this.get("_listeners", activation, context)?;
+
+    if let Value::Object(listeners) = listeners {
+        let length = listeners.length();
+        let mut position = None;
+
+        for i in 0..length {
+            let other_listener = listeners.array_element(i);
+            if new_listener == other_listener {
+                position = Some(i);
+                break;
+            }
+        }
+
+        if position == None {
+            listeners.set_length(context.gc_context, length + 1);
+            listeners.set_array_element(length, new_listener, context.gc_context);
+        }
+    }
+
+    Ok(true.into())
+}
+
+pub fn remove_listener<'gc>(
+    activation: &mut Activation<'_, 'gc>,
+    context: &mut UpdateContext<'_, 'gc, '_>,
+    this: Object<'gc>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error<'gc>> {
+    let old_listener = args.get(0).cloned().unwrap_or(Value::Undefined);
+    let listeners = this.get("_listeners", activation, context)?;
+
+    let mut removed = false;
+    if let Value::Object(listeners) = listeners {
+        let length = listeners.length();
+        let mut position = None;
+
+        for i in 0..length {
+            let other_listener = listeners.array_element(i);
+            if old_listener == other_listener {
+                position = Some(i);
+                break;
+            }
+        }
+
+        if let Some(position) = position {
+            if length > 0 {
+                let new_length = length - 1;
+                for i in position..new_length {
+                    listeners.set_array_element(
+                        i,
+                        listeners.array_element(i + 1),
+                        context.gc_context,
+                    );
+                }
+
+                listeners.delete_array_element(new_length, context.gc_context);
+                listeners.delete(activation, context.gc_context, &new_length.to_string());
+
+                listeners.set_length(context.gc_context, new_length);
+
+                removed = true;
+            }
+        }
+    }
+
+    Ok(removed.into())
+}
+
+pub fn broadcast_message<'gc>(
+    activation: &mut Activation<'_, 'gc>,
+    context: &mut UpdateContext<'_, 'gc, '_>,
+    this: Object<'gc>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error<'gc>> {
+    let event_name_value = args.get(0).cloned().unwrap_or(Value::Undefined);
+    let event_name = event_name_value.coerce_to_string(activation, context)?;
+    let call_args = &args[1..];
+
+    broadcast_internal(activation, context, this, call_args, &event_name)
```

Comment:
This panics if you call `broadcastMessage()` with no params. Try `call_args.get()`. Additionally it looks like calling with no params should bail out and do nothing.